pacsman/dcmtk_client.py

```diff
@@ -483,3 +528,37 @@ def send_datasets(self, datasets: Iterable[Dataset], override_remote_ae: str = N
                     msg = f'Failure to send dataset with {dataset.SeriesInstanceUID}, rc {result.returncode}'
                     logger.error(msg)
                     raise Exception(msg)
+
+
+def _check_dcmtk_message_for_error(dcmtk_message: str) -> Optional[Tuple[int, int]]:
+    """
+    This checks a message from DCMTK for a known error message pattern.
+
+    :param dcmtk_message: The message to parse / check for an error message.
+    :return: Either `None` if no error is detected, else a tuple with (module_code, err_code)
+
+    This is useful, because for certain operations, DCMTK is likely to always return a zero exit code, even in cases
+    of (nested) failure.
+
+    For example, `findscu` calls will seemingly only return a non-zero exit code if the failure happens at the
+    association level (e.g. on setup or abort).
+
+    This is a known issue: https://support.dcmtk.org/redmine/issues/929
+    """
+    # Standard pattern is E: (0x0001, 0x0002) ERROR MESSAGE
```

Comment:
Based on your regex and the example snippets posted here, it looks like the standard pattern is `E: 0001:0002 ERROR MESSAGE`, right? If so I think we should update this comment.

Reply:
Good catch! Updated.